src/sentry/api/validators/project_codeowners.py

```diff
@@ -21,18 +20,20 @@
     from sentry.services.hybrid_cloud.user import RpcUser
 
 
-def validate_association(
-    raw_items: Sequence[UserEmail | ExternalActor],
-    associations: Sequence[UserEmail | ExternalActor],
-    type: str,
-) -> Sequence[str]:
+def validate_association_emails(
+    raw_items: Collection[str],
+    associations: Collection[str],
+) -> list[str]:
+    return list(set(raw_items).difference(associations))
+
+
+def validate_association_actors(
+    raw_items: Collection[str],
+    associations: Collection[ExternalActor],
+) -> list[str]:
     raw_items_set = {str(item) for item in raw_items}
-    if type == "emails":
-        # associations email strings
-        sentry_items = associations
-    else:
-        # associations are ExternalActor objects
-        sentry_items = {item.external_name for item in associations}
+    # associations are ExternalActor objects
+    sentry_items = {item.external_name for item in associations}
     return list(raw_items_set.difference(sentry_items))
```

Comment:
this could have been done with `@overload` -- but I felt the two separate functions communicated the intent better